module/applications/api/primary-sheet-mixin.mjs

```diff
@@ -0,0 +1,440 @@
+     * @param {Event} event         Triggering click event.
+     * @param {HTMLElement} target  Button that was clicked.
+     * @returns {any}
+     * @protected
+     * @abstract
+     */
+    _addDocument(event, target) {}
+
+    /* -------------------------------------------- */
+
+    /**
+     * Initialize item tooltips on an element.
+     * @param {HTMLElement} element  The tooltipped element.
+     * @protected
+     */
+    _applyItemTooltips(element) {
+      if ( "tooltip" in element.dataset ) return;
+      const target = element.closest("[data-item-id], [data-effect-id], [data-uuid]");
+      let uuid = target.dataset.uuid;
+      if ( !uuid && target.dataset.itemId ) {
+        const item = this.actor?.items.get(target.dataset.itemId);
+        uuid = item?.uuid;
+      } else if ( !uuid && target.dataset.effectId ) {
+        const { effectId, parentId } = target.dataset;
+        const collection = parentId ? this.actor?.items.get(parentId).effects : this.actor?.effects;
+        uuid = collection?.get(effectId)?.uuid;
+      }
+      if ( !uuid ) return;
+      element.dataset.tooltip = `
+        <section class="loading" data-uuid="${uuid}"><i class="fas fa-spinner fa-spin-pulse"></i></section>
+      `;
+      element.dataset.tooltipClass = "dnd5e2 dnd5e-tooltip item-tooltip";
+      element.dataset.tooltipDirection ??= "LEFT";
+    }
+
+    /* -------------------------------------------- */
+
+    /** @inheritDoc */
+    changeTab(tab, group, options) {
+      super.changeTab(tab, group, options);
+      if ( group !== "sheet" ) return;
+      this.element.className = this.element.className.replace(/tab-\w+/g, "");
+      this.element.classList.add(`tab-${tab}`);
+    }
+
+    /* -------------------------------------------- */
+
+    /**
+     * Handle the user toggling the sheet mode.
+     * @param {Event} event  The triggering event.
+     * @protected
+     */
+    async _onChangeSheetMode(event) {
+      const { MODES } = this.constructor;
+      const toggle = event.currentTarget;
+      const label = game.i18n.localize(`DND5E.SheetMode${toggle.checked ? "Play" : "Edit"}`);
+      toggle.dataset.tooltip = label;
+      toggle.setAttribute("aria-label", label);
+      this._mode = toggle.checked ? MODES.EDIT : MODES.PLAY;
+      await this.submit();
+      this.render();
+    }
+
+    /* -------------------------------------------- */
+
+    /** @inheritDoc */
+    _onClickAction(event, target) {
+      if ( target.dataset.action === "addDocument" ) this._addDocument(event, target);
+      else if ( target.dataset.action === "editImage" ) this._editImage(event, target);
+      else super._onClickAction(event, target);
+    }
+
+    /* -------------------------------------------- */
+
+    /**
+     * Handle toggling an Item's description within an inventory list.
+     * @param {PointerEvent} event  The triggering event.
+     * @protected
+     */
+    async _onToggleDescription(event) {
```

Comment:
I thought the PR description suggested this was going to be merged with the collapse/expand code from the activity sheet, but it seems that's not happening here. I was thinking of maybe moving this responsibility to the `InventoryElement`. I've moved in that direction on my prototype branch so if that sounds reasonable, we can pull this code out here and wait for the `InventoryElement` changes.

Reply:
That mention was for the main descriptions on the item sheets, not the item summaries in the inventory area. Because this needs to perform the extra fetching and rendering it couldn't be merged with the other code. Inventory element would probably be the best place for this, yeah.

Reply:
> That mention was for the main descriptions on the item sheets, not the item summaries in the inventory area.

Ah ok, that makes sense.

> When this ends up in `InventoryElement`, it would be good to take advantage of `expandedSections` since it will be available in all of our AppV2 sheets rather than tracking those separately.

Will keep this in mind. Certainly remind me again if it doesn't end up in the PR.